src/main/java/redis/clients/jedis/JedisClientSideCache.java

```diff
@@ -0,0 +1,45 @@
+package redis.clients.jedis;
+
+import redis.clients.jedis.exceptions.JedisException;
+
+public class JedisClientSideCache extends Jedis {
+
+  private final ClientSideCache cache;
+
+  public JedisClientSideCache(final HostAndPort hostPort, final JedisClientConfig config) {
+    this(hostPort, config, new ClientSideCache());
+  }
+
+  public JedisClientSideCache(final HostAndPort hostPort, final JedisClientConfig config,
+      ClientSideCache cache) {
+    super(hostPort, config);
+    if (config.getRedisProtocol() != RedisProtocol.RESP3) {
```

Comment:
@vladvildanov @dvora-h IMHO we should do this in other clients if not already there.

Reply:
@chayim It's already there

But, we had a discussion with you and @uglide and we agreed that we don't throw an exception in case if user do not explicitly set RESP3, but instead we're forcing RESP3 connection.